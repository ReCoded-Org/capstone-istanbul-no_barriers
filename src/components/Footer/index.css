/* color variables */
:root {
  --main-footer-background-colour: #002736;
  --text: #eff4f6;
  --hover-text: #f2c94c;
}

.footer {
  background-color: var(--main-footer-background-colour);
  position: relative;
  padding-bottom: 1.5rem;
  text-align: center;
}

.footerNavbar {
  background-color: var(--main-footer-background-colour);
}

/* links to pages */
.navbar-expand .navbar-nav .nav-link {
  padding-left: 2.5rem;
}

/* No barriers main link */
.navbar-dark .navbar-brand {
  padding-top: 1.1rem;
}

.footerHeading {
  padding-right: 0.5rem;
}

.copyright {
  font-size: 1rem;
  padding-top: 0.35rem;
}

/* all links style */
.navbar-dark .navbar-brand,
.privacyTerms,
.navbar-expand .navbar-nav .nav-link {
  color: var(--text);
}

/* all links on hover */
.navbar-dark .navbar-brand:hover,
.privacyTerms:hover,
.navbar-expand .navbar-nav .nav-link:hover {
  color: var(--hover-text);
}

.privacyTerms {
  position: absolute;
  width: 10rem;
  top: -2rem;
  left: -1rem;
  font-size: smaller;
}

.privacyTerms:hover {
  text-decoration: none;
}
