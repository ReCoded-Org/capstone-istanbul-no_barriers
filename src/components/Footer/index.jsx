import React from "react";
import { useTranslation } from "react-i18next";
import { Container, Navbar, Nav, Col } from "react-bootstrap";
import "./index.css";

const Footer = () => {
  const { t } = useTranslation();

  return (
    <div className="footer">
      <Container fluid>
        <Navbar
          variant="dark"
          className="footerNavbar"
          collapseOnSelect="false"
          expand="lg"
        >
          <Navbar.Brand href="/" className="d-flex">
            <h4 className="footerHeading">{t("homePage.footer.brand")}</h4>
            <p className="copyright">
              <span>&#169;</span>
              2020
            </p>
          </Navbar.Brand>
          <Nav className="ml-auto">
            <Nav.Link href="/findngos">{t("homePage.footer.find")}</Nav.Link>
            <Nav.Link href="/stories">{t("homePage.footer.stories")}</Nav.Link>
            <Nav.Link href="/aboutUs">{t("homePage.footer.about")}</Nav.Link>
          </Nav>
        </Navbar>
        <Col className="justify-content-md-left" md={2}>
          <a href="#term" target="blank" className="privacyTerms">
            {t("homePage.footer.term")}
          </a>
        </Col>
      </Container>
    </div>
  );
};

export default Footer;
