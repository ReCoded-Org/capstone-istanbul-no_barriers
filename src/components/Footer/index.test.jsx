import React from "react";
import renderer from "react-test-renderer";
import Footer from "./index";

it("Footer renders correctly", () => {
  const tree = renderer.create(<Footer />).toJSON();
  expect(tree).toMatchSnapshot();
});
