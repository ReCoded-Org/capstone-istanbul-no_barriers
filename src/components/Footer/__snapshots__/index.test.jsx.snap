// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer renders correctly 1`] = `
<div
  className="footer"
>
  <div
    className="container-fluid"
  >
    <nav
      className="footerNavbar navbar navbar-expand navbar-dark"
    >
      <a
        className="d-flex navbar-brand"
        href="/"
      >
        <h4
          className="footerHeading"
        >
          homePage.footer.brand
        </h4>
        <p
          className="copyright"
        >
          <span>
            Â©
          </span>
          2020
        </p>
      </a>
      <div
        className="ml-auto navbar-nav"
        onKeyDown={[Function]}
      >
        <a
          className="nav-link"
          data-rb-event-key="/findNgos"
          href="/findNgos"
          onClick={[Function]}
          onKeyDown={[Function]}
        >
          homePage.footer.find
        </a>
        <a
          className="nav-link"
          data-rb-event-key="/stories"
          href="/stories"
          onClick={[Function]}
          onKeyDown={[Function]}
        >
          homePage.footer.stories
        </a>
        <a
          className="nav-link"
          data-rb-event-key="/aboutUs"
          href="/aboutUs"
          onClick={[Function]}
          onKeyDown={[Function]}
        >
          homePage.footer.about
        </a>
      </div>
    </nav>
    <div
      className="justify-content-md-left col-md-2"
    >
      <a
        className="privacyTerms"
        href="#term"
        target="blank"
      >
        homePage.footer.term
      </a>
    </div>
  </div>
</div>
`;
